### ISSUE TITLE
OpenSSL 1.0.2 'make clean' doesn't clean generated .S files

### ISSUE THREAD

**Issue by JonathanLennox** [openssl/openssl#2220]
When building for both the linux-armv4 and linux-aarch64 targets, OpenSSL 1.0.2j's make process generates .S files for ARM assembly, in the crypto/aes, crypto/bn, crypto/modes, and crypto/sha directories.

Unfortunately, 'make clean' does not remove these files, and re-running Configure or 'make depend' does not do anything to let make know they need to be regenerated.

The syntax of the generated files is different for 32-bit and 64-bit arm.

Thus, if you try to build OpenSSL for both a 32-bit arm and 64-bit arm platform in the same OpenSSL directory, the second build will fail.

The solution is to add *.S to the 'clean' target in those directories.

**Comment by bernd-edlinger** [comment]
I can reproduce this.
A grep reveals that also crypto/md5, crypto/rc4 and crypto/des may contain *.S files:

```
grep -r "\.S:" crypto
crypto/modes/Makefile:ghashv8-armx.S:	asm/ghashv8-armx.pl
crypto/modes/Makefile:ghash-%.S:	asm/ghash-%.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
crypto/md5/Makefile:md5-sparcv9.S:	asm/md5-sparcv9.pl
crypto/rc4/Makefile:rc4-ia64.S: asm/rc4-ia64.pl
crypto/sha/Makefile:sha256-armv4.S: asm/sha256-armv4.pl
crypto/sha/Makefile:sha1-sparcv9.S:	asm/sha1-sparcv9.pl;	$(PERL) asm/sha1-sparcv9.pl $@ $(CFLAGS)
crypto/sha/Makefile:sha256-sparcv9.S:asm/sha512-sparcv9.pl;	$(PERL) asm/sha512-sparcv9.pl $@ $(CFLAGS)
crypto/sha/Makefile:sha512-sparcv9.S:asm/sha512-sparcv9.pl;	$(PERL) asm/sha512-sparcv9.pl $@ $(CFLAGS)
crypto/sha/Makefile:sha1-mips.S:	asm/sha1-mips.pl;	$(PERL) asm/sha1-mips.pl $(PERLASM_SCHEME) $@
crypto/sha/Makefile:sha256-mips.S:	asm/sha512-mips.pl;	$(PERL) asm/sha512-mips.pl $(PERLASM_SCHEME) $@
crypto/sha/Makefile:sha512-mips.S:	asm/sha512-mips.pl;	$(PERL) asm/sha512-mips.pl $(PERLASM_SCHEME) $@
crypto/sha/Makefile:sha1-%.S:	asm/sha1-%.pl;		$(PERL) $< $(PERLASM_SCHEME) $@
crypto/sha/Makefile:sha256-%.S:	asm/sha512-%.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
crypto/sha/Makefile:sha512-%.S:	asm/sha512-%.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
crypto/des/Makefile:des_enc-sparc.S:	asm/des_enc.m4
crypto/aes/Makefile:aes-mips.S:	asm/aes-mips.pl
crypto/aes/Makefile:aesv8-armx.S:	asm/aesv8-armx.pl
crypto/aes/Makefile:aes-%.S:	asm/aes-%.pl;	$(PERL) $< $(PERLASM_SCHEME) > $@
crypto/aes/Makefile:bsaes-%.S:	asm/bsaes-%.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
crypto/bn/Makefile:sparct4-mont.S:	asm/sparct4-mont.pl
crypto/bn/Makefile:sparcv9-gf2m.S:	asm/sparcv9-gf2m.pl
crypto/bn/Makefile:%-mont.S:	asm/%-mont.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
crypto/bn/Makefile:%-gf2m.S:	asm/%-gf2m.pl;	$(PERL) $< $(PERLASM_SCHEME) $@
```

**Comment by richsalz** [comment]
fixed with 9b90ce0.  thanks for reporting this!
